Make our docs guidance discoverable

ptgott opened the pull request:
Currently, our guidance for making docs changes is
organized into a /docs/docs page that is only accessible if
you know the URL. This change ensures that people who have not
received an informal training on docs contributions can
contribute in a way that abides by our expectations for
consistency and effectiveness.

- Splits the docs/docs and docs/docs/best-practices pages into
  a UI reference, getting started guide, and style guide. This
  ensures that each kind of content we have in our contributing
  guide (i.e., tutorial or reference) is accessible from the
  main nav menu.
- Adds a "Contributing" tab to the navigation menu. There is
  one subsection here related to documenation, and we can add
  another subsection later with developer guides for editing
  Teleport's code.
- Moves the "philosophy" section into the intro page of the
  documentation contribution section menu.
- Makes various edits to the original docs guidance for clarity.

Fixes #9538

ptgott committed:
Make our docs guidance discoverable

Currently, our guidance for making docs changes is
organized into a /docs/docs page that is only accessible if
you know the URL. This change ensures that people who have not
received an informal training on docs contributions can
contribute in a way that abides by our expectations for
consistency and effectiveness.

- Splits the docs/docs and docs/docs/best-practices pages into
  a UI reference, getting started guide, and style guide. This
  ensures that each kind of content we have in our contributing
  guide (i.e., tutorial or reference) is accessible from the
  main nav menu.
- Adds a "Contributing" tab to the navigation menu. There is
  one subsection here related to documenation, and we can add
  another subsection later with developer guides for editing
  Teleport's code.
- Moves the "philosophy" section into the intro page of the
  documentation contribution section menu.
- Makes various edits to the original docs guidance for clarity.
- Document our docs labels in a dedicated "creating an issue"
  page

Fixes #9538

ptgott commented:
One thing we can also consider is splitting the "style-guide" page so that the "Kinds of pages" and "General style rules" H2 sections are separate pages.

zmb3 commented on docs/pages/contributing/documentation.mdx:
```diff
@@ -0,0 +1,27 @@
+layout: tocless-doc
+---
+
+Documentation is essential for learning about a product or service. Even for internal contributors, reference documentation is often the ultimate source of truth and the final arbiter of disputes and disagreements. And for decision makers at organizations evaluating solutions, documentation is often a key resource.
```
nit:

I typically prefer to hard wrap lines at a reasonable point rather than have really long lines. It's easier to read in an editor, and also easier to review and comment on individual lines rather than paragraphs.

There's lots of editor plugins that can do this automatically. I prefer `reflow-markdown` for vs-code.

zmb3 commented on docs/pages/contributing/documentation.mdx:
Any reason not to use a markdown list instead of `<ul>`?

zmb3 commented on docs/pages/contributing/documentation/how-to-contribute.mdx:
````diff
@@ -0,0 +1,92 @@
+
+```code
+$ git fetch origin
+$ git checkout branch/v8
````
Should we recommend the more modern git commands for this? `git switch` exists because `git checkout` is confusing to new users.

```
$ git switch branch/v8
$ git switch -c my-branch-backport-v8
$ git cherry-pick ...

```

zmb3 commented on docs/pages/contributing/documentation/style-guide.mdx:
```diff
@@ -0,0 +1,153 @@
+
+### Lists
+- We prefer short paragraph blocks over bulleted or numbered lists. This leads to a preference for completeness and brevity rather than enumeration. 
+- When you do include a list, we prefer bullet points over numberd lists.
```
```suggestion
- When you do include a list, we prefer bullet points over numbered lists.
```

zmb3 commented on docs/pages/contributing/documentation/style-guide.mdx:
```diff
@@ -0,0 +1,153 @@
+### Lists
+- We prefer short paragraph blocks over bulleted or numbered lists. This leads to a preference for completeness and brevity rather than enumeration. 
+- When you do include a list, we prefer bullet points over numberd lists.
+- Use numbered lists for any sequence of steps, but use these sparingly.
```
IIRC, @klizhentas prefers `1/N` notation for a sequence of steps, so readers know how far they are in the process. Do we want to mention that here?

ptgott committed:
Respond to PR feedback

- Use a Markdown list for the section index page
- Break up lines
- Use `git switch` instead of `git checkout` in the git workflow
  example
- Advise using denominators in numbered lists

ptgott committed:
Fix formatting
